File: Dia01/d1p1/src/main.js
readFileSync = require('fs').readFileSync;

let data = readFileSync('../../input.txt', 'utf8').split('\n').map((v, i, arr) => parseInt(v.trim()));
let sum = 0;
for (let i = 1; i < data.length; i++) {
	if (data[i - 1] < data[i])
		sum++;
}

console.log(sum);

File: Dia01/d1p1/src/main.rs
fn main() {
    let r = include_str!("../../input.txt")
        .lines()
        .map(|x| x.parse::<usize>().unwrap())
        .fold((None, 0), |(last, sum), n| {
            (
                Some(n),
                if last.map(|x| x < n).unwrap_or(false) {
                    1
                } else {
                    0
                } + sum,
            )
        })
        .1;

    println!(">> {}", r);
}
